conduct.md

```diff
@@ -24,15 +21,15 @@ As a member of the community, you agree to the following:
-- Remember that everyone was new to Scala at some point. We want to encourage newcomers to join our community and learn the Scala language and ecosystem. Assume competence.
-- Show empathy towards other community members.
+- **Be kind and courteous.** We treat our fellow community members with the empathy, respect and dignity all humans deserve. Keep in mind that public communication is received by many people you don’t know, so before sending a message please ask yourself whether someone from a different context would misunderstand it.
+- **Respect differences of opinion** and remember that every design or implementation choice carries a trade-off and numerous costs. There is seldom a single right answer; we will find the best solutions by engaging in constructive discussion, with everybody bringing their unique viewpoint and experience to the table.
+- **Remember that everyone was new to Scala at some point.** We want to encourage newcomers to join our community and learn the Scala language and ecosystem. Always assume good intentions and a willingness to learn, just as you are willing to evolve your own opinion as you gain new insights.
 
 **Discouraged:**
 
+- Do not exclude people or reject their contributions based on age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, sexual identity and orientation, or other such characteristics.
 - Keep unstructured critique to a minimum. We encourage sharing ideas and perspectives, so please ensure that your feedback is constructive and relevant. If you have solid ideas you want to experiment with, make a fork and see how it works.
-- Avoid aggressive and micro-aggressive behavior, such as unconstructive criticism, providing corrections that do not improve the conversation (sometimes referred to as "well actually"s), repeatedly interrupting or talking over someone else, feigning surprise at someone’s lack of knowledge or awareness about a topic, or subtle prejudice (for example, comments like “That’s so easy my grandmother could do it.”). For more examples of this kind of behavior, [see the Recurse Center's user manual](https://www.recurse.com/manual#sec-environment).
+- Avoid behaviors that lead to strife and tension, such as unconstructive criticism, providing corrections that do not improve the conversation (sometimes referred to as "well actually"s), repeatedly interrupting or talking over someone else, feigning surprise at someone’s lack of knowledge or awareness about a topic, or subtle prejudice (for example, comments like “That’s so easy my grandmother could do it.”). For more examples of this kind of behavior, [see the Recurse Center's user manual](https://www.recurse.com/manual#sec-environment).
```

Comment:
This change alters the meaning in a way I don't think was intended. "Micro-aggressive" is meant to describe subtle exclusion, whereas "strife and tension" might be interpreted to mean active conflict. That could lead people to think that as long as some behavior doesn't cause a flame war to break out, it's fine. We want people to also think about how their words and behavior might cause other problems such as people simply choosing not to engage in the community anymore because they were treated disrespectfully by other members.

I appreciate that many people might not be familiar with the term "microaggression". What does everyone think about this wording? "Avoid aggressive, condescending, or dismissive behavior..."

Reply:
As I mentioned before, I think it's easier to police overall mood (strife, tension) than individual actions that depend in large part on the (difficult-to-know) intent of the speaker.  But I think "aggressive, condescending, or dismissive" is acceptable too.

Also, strife is active, but tension isn't, particularly.  How likely is it, do you think, that people would think it only applied to active conflict, especially given the examples?  Maybe the two of those are not enough, though; one can be excluded in ways that are low-strife and not very tense, but still very exclusionary (e.g. "cold shoulder").  On the other hand, every example causes either strife or tension or both.

I don't think unfamiliarity with the term "micro-aggression" is the problem with using the term.  The problem is that (1) it carries a connotation among some, deservedly or not, that it is expected that people will go into hysterics over a single tiny unintended or unknowable perceived slight; and (2) at least in my opinion it is implicitly demeaning to the targets of condescending etc behavior by suggesting that what's happening to them is "micro" (even though the originators of the term AFAICT intended the opposite).  Social exclusion and status-lowering behaviors can feel really bad (comparable to moderate levels of physical aggression, or people wouldn't be willing e.g. to get into fights over it, which they are), which is entirely reasonable given how critical social standing is to social creatures like us.  Anyway, we have lots of other better-established words that can convey pretty much anything we want to convey, both in literal meaning and in implication.  The only thing we'd miss is, "Hey, we're hip to this new terminology that's in fashion now."

Reply:
I like Rex's unpacking of what's less-than-ideal about using the term "micro-aggression", thanks for that. So I think a good replacement wording is still worth trying for.

I had initially liked the "Avoid behaviors that lead to strife and tension" wording, but Tim's remarks give me pause.

maybe a slight rewording to "tension or strife" would help improve that version?

@TimMoore your version isn't bad, but it changes the emphasis in a way I think is probably not desirable. in "tension or strife", the emphasis is on the effect behavior has on others. But "aggressive, condescending, or dismissive" puts the emphasis on the behavior in itself, and I think that could encourage arguing about the definitions. It's better during enforcement to say things like "you're bothering others" than "your behavior was aggressive"; people will more readily accept the former, whereas the latter puts them on the defensive, in my experience.

Reply:
@SethTisue I like your proposal of “tension or strife”, also because it is listed in escalating order.

One issue (perhaps the main one bothering me and others over here) is that “micro-aggression” is a foreign concept in Europe AFAIK — we tend to use established words to describe the effects. This makes me uncomfortable because I do not have a good intuition for what “micro-aggression” really means.

Reply:
@SethTisue: good points. I agree that the motivation here should be to emphasize the effects on others in the community.

@rkuhn: also good points. I think it's fair to say that the term is probably not widely known anywhere, and people who do understand the term are likely to be more self-aware about it already. I acknowledge that it can be a controversial term, and can easily be misinterpreted, so I'm positive on being more explicit about the intent.

One thing that I like about this code of conduct is that it isn't only about "policing" egregious harm by people with bad intent, but also about encouraging members of the community to be aware of more subtle forms of exclusion and to take an active role in creating a more welcoming community.

People who have suffered a history of exclusion might approach a new community by looking for patterns of communication or behavior that fit their previous bad experiences. If they see indications of this, they might immediately turn away from the community, having no more patience for repeating that history. This is a bad outcome for the community. Does it qualify as tension or strife? I think we're missing something that captures that nuance. Maybe "marginalization, tension or strife"?

Unfortunately, @Ichoran I think your comment mischaracterizes the intent of the concept of microaggression and in doing so demonstrates some of the problem. From what I understand, the original intent was to talk about incidents that are individually small enough that those affected think "it's not worth making a big deal out of it", but have a cumulative effect of making them feel like second-class citizens. These problems are rarely raised to attention, especially because the people affected by them expect to be dismissed, or further stigmatized, if they do. Even the phrase "go into hysterics" is loaded with a history of exaggerated and discriminatory use that some people will interpret as a cue that reporting small problems in the community will be unwelcome, disrespected, and provoke a defensive overreaction. I would also say that "Hey, we're hip to this new terminology that's in fashion now" could be more generously interpreted as "Hey, we're educated about and sensitive to common causes of subtle harm that have been raised by people who have experienced them."

Reply:
@TimMoore - I'm intentionally mischaracterizing the intent of the concept of micro-aggression because intent is not the only thing that determines how language is understood; among other things, the literal meaning, the cultural context, and the existing implicit associations of the words used, also have a major impact on how languages convey meaning.

People have been marginalizing, dismissing, mocking, teasing, stigmatizing, goading, maligning, negatively stereotyping, demeaning, disrespecting, etc. etc. each other for a very long time, and we consequently have a rich vocabulary with which to describe all manner of inappropriate and/or negative treatment.  The thing that "micro-aggression" adds to the vocabulary which is possibly useful is a general term to describe the category of individual acts that in aggregate cause a substantially negative experience.

However, it's not clear to me that there's any benefit for us to talk about the individual acts as opposed to the overall impact, and there are drawbacks to the terminology: unfamiliarity, and connotations that the term carries for some that we probably don't actually intend.  If we're talking about overall impact, we don't need the kind of generified quantization that "micro-aggression" provides.

Anyway, I think for now we should put the beginning of the document back _exactly the way it was_ and discuss the content of the list and the phrasing of the first few sentences in another place, so as not to indefinitely postpone the other helpful changes in this PR.

Reply:
I’ve reverted this change as well, to be improved in a follow-up PR.